tenantcostcontrol: change settings to TenantReadOnly

RaduBerinde opened the pull request:
#### tenant settings: implement TenantReadOnly semantics

This commit implements the advertised `TenantReadOnly` semantics,
changes the `kv.protectedts.reconciliation.interval` setting to
`TenantReadOnly`, and adds some relevant tests.

Note that the settings watcher already ignores any values for
`TenantReadOnly` variables in the `system.settings` table. This commit
includes a test for that.

Release note: None

#### tenantcostcontrol: change settings to TenantReadOnly

This commit changes the settings relevant for RU accounting and
limiting to `TenantReadOnly`. The tenant cost model values are now
used (instead of always using the default values). The range of values
for `tenant_cpu_usage_allowance` is extended since we no longer have
to worry about the tenant bumping it up.

Release note: None

RaduBerinde committed:
tenantcostcontrol: change settings to TenantReadOnly

This commit changes the settings relevant for RU accounting and
limiting to `TenantReadOnly`. The tenant cost model values are now
used (instead of always using the default values). The range of values
for `tenant_cpu_usage_allowance` is extended since we no longer have
to worry about the tenant bumping it up.

Release note: None

RaduBerinde committed:
tenant settings: implement TenantReadOnly semantics

This commit implements the advertised `TenantReadOnly` semantics,
changes the `kv.protectedts.reconciliation.interval` setting to
`TenantReadOnly`, and adds some relevant tests.

Note that the settings watcher already ignores any values for
`TenantReadOnly` variables in the system.settings table. This commit
includes a test for that.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/76680)
<!-- Reviewable:end -->

ajwerner reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 3 of 3 files at r1, 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/76680#-:-Mw2AC0A219ULe021IwE:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @ajstorm and @dt)



<!-- Sent from Reviewable.io -->

RaduBerinde committed:
jobsprotectedtsccl: fix use of settings tests

This commit fixes these tests to set these settings on the host
cluster as well as on the tenant.

Release note: None

RaduBerinde commented:
I added a commit that fixes the use of settings in the jobsprotectedts tests. CC @adityamaru

RaduBerinde commented:
bors r+
